### ISSUE TITLE
🐛 --reporter=summary merging rules and giving incorrect issue counts

### ISSUE THREAD

**Issue by pthurston** [biomejs/biome#8730]
### Environment information

<details>

```bash

CLI:
  Version:                      2.3.11
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.19.6
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/10.8.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false
  HTML full support enabled:    unset

Workspace:
  Open Documents:               0

```

</details>

### What happened?

I've noticed that if `biome lint` or `biome check` are run with `--reporter=summary` then some diagnostic counts get inappropriately attributed to the wrong rule. 

Codesandbox link: https://codesandbox.io/p/devbox/clever-colden-9fq8xk
I added a "summary" task to reproduce the issue.

In my example I have a test file that has 8 issues (4 noUnusedVariables, 1 noGlobalIsNan, and 3 noExplicitAny).

The issue is that given this test file, the summary reporter outputs this:
```bash
  Rule Name                                        Diagnostics
  
  lint/correctness/noUnusedVariables               4 (4 warnings)
  lint/suspicious/noGlobalIsNan                         4 (4 warnings)
```
which for some reason is grouping the noExplicitAny under noGlobalIsNan. 

I also tested out changing the noExplicitAny rule to the "info" level, which resulted in this summary:
```bash
  Rule Name                                        Diagnostics
  
  lint/correctness/noUnusedVariables               4 (4 warnings)
  lint/suspicious/noGlobalIsNan                         4 (1 warning, 3 infos)
```
Which is why I'm confident the 3 extra diagnostics attributed to noGlobalIsNan are from noExplicitAny. 

### Expected result

I would expect the summary output to be something like this
```bash
  lint/correctness/noUnusedVariables               4 (4 warnings)
  lint/suspicious/noGlobalIsNan                         1 (1 warning)
  lint/suspicious/noExplicitAny                          3 (3 warnings)
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
